app/uk/gov/hmrc/soletraderidentificationfrontend/services/SubmissionService.scala

```diff
@@ -25,61 +25,163 @@ import javax.inject.{Inject, Singleton}
-        } yield {
-          JourneyCompleted(journeyConfig.continueUrl)
-        }
-        case Right(_) => for {
-          _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
-          _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
-        } yield {
-          JourneyCompleted(journeyConfig.continueUrl)
-        }
-        case Left(failureReason) =>
+    } yield
+      response
+
+  private def noDataServerException(journeyId: String): Nothing =
+    throw new InternalServerException(s"Details could not be retrieved from the database for $journeyId")
+
+  private def calculateBusinessScenario(journeyConfig: JourneyConfig): BusinessScenario =
+    if (!journeyConfig.pageConfig.enableSautrCheck) IndividualJourney
+    else {
+      if (journeyConfig.businessVerificationCheck) SoleTraderJourneyWithSaUtrCheck
+      else SoleTraderJourneySkippingSaUtrCheck
+    }
+
+  private def calculateSubmissionResponse(journeyId: String,
+                                          businessScenario: BusinessScenario,
+                                          matchingResult: Either[SoleTraderDetailsMatching.SoleTraderDetailsMatchFailure, Boolean],
+                                          journeyConfig: JourneyConfig,
+                                          individualDetails: IndividualDetails)(implicit hc: HeaderCarrier,
+                                                                                ec: ExecutionContext): Future[SubmissionResponse] = businessScenario match {
+    case SoleTraderJourneySkippingSaUtrCheck =>
+      handleSoleTraderJourneySkippingBVCheck(journeyId, matchingResult, journeyConfig.continueUrl, individualDetails.optSautr, individualDetails.optNino)
+    case SoleTraderJourneyWithSaUtrCheck =>
+      handleSoleTraderJourneyWithBVCheck(journeyId, matchingResult, journeyConfig, individualDetails)
+    case IndividualJourney =>
+      handleIndividualJourney(journeyId, matchingResult, journeyConfig.continueUrl)
+  }
+
+  private def handleSoleTraderJourneySkippingBVCheck(journeyId: String,
+                                                     matchingResult: Either[SoleTraderDetailsMatching.SoleTraderDetailsMatchFailure, Boolean],
+                                                     continueUrl: String,
+                                                     maybeSaUtr: Option[String],
+                                                     maybeNino: Option[String]
+                                                    )
+                                                    (implicit hc: HeaderCarrier,
+                                                     ec: ExecutionContext): Future[SubmissionResponse] = matchingResult match {
+    case Right(true) if maybeSaUtr.isDefined =>
+      registrationOrchestrationService
+        .registerWithoutBusinessVerification(journeyId, maybeNino, maybeSaUtr.getOrElse(aSaUtrBug))
```

Comment:
Do we need aSaUtrBug? We are testing that the option is defined and so "get" should work fine. In the unlikely scenario that maybeSaUtr was undefined "get" would throw an exception anyway.

Reply:
.... yes Richard I wrote it just to put more information in case things goes wrong ... just to create a context that in case something goes wrong give us more info about where we were .... if we use a simple get we will have an anonymously "
java.util.NoSuchElementException: None.get
  at scala.None$.get(Option.scala:627)
"
exception that does not give the impression we were aware of it ..... but if we have  something like 
java.lang.IllegalStateException: This is a bug. sa-utr should be Defined but it is not!
  at $anonfun$res0$1(<console>:1)
  at scala.Option.getOrElse(Option.scala:201)
  ... 40 elided
it seems more cool ;-) ... joking it seems better .... personal taste ... happy to change